Setup Liquibase [POL-30]

snwessel opened the pull request.

zarsky-broad committed:
initial schema

zarsky-broad committed:
update schema to match updated doc

Co-Authored-By: Kai <19541449+Shakespeared@users.noreply.github.com>
Co-Authored-By: Sarah Wessel <23193756+snwessel@users.noreply.github.com>

zarsky-broad committed:
no extra lines

dvoet commented on src/main/resources/db/changelog/changesets/21052021_initial_schema.yaml:
```diff
@@ -0,0 +1,109 @@
+      author: kfeldman, swessel, zarsky
+      changes:
+        - sql:
+            sql: "CREATE TYPE token_type_enum AS ENUM ('document', 'code')"
```
I would call this `document_token` and `access_token` (or just `document` and `access` but plain `access` feels weird)

dvoet commented on src/main/resources/db/changelog/changesets/21052021_initial_schema.yaml:
```diff
@@ -0,0 +1,109 @@
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: user_id
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: provider_id
+                  type: text
+                  constraints:
+                    nullable: false
```
need a unique constraint on these 2 together

snwessel committed:
Add unique constraint, update enum types

snwessel commented:
I just updated the enum type names and added the unique constraint as suggested. When I ran the app, it created the constraint successfully! 
<img width="514" alt="image" src="https://user-images.githubusercontent.com/23193756/120342064-6db73080-c2c5-11eb-95da-02d923fb68ee.png">
